### ISSUE TITLE
Configure pylint or replace it

### ISSUE THREAD

**Issue by roll** [okfn/coding-standards#9]
I've switched `tabulator` from `pylama` to `pylint` as we've decided but now I'm having some problems with it. For example:

```
$ pylama tabulator - OK
$ pep8 tabulator - OK
$ pyflakes tabulator - OK
$ flake8 tabulator - OK
```

But `pylint` finishes with many message like:

```
************* Module tabulator.iterator
R: 12, 0: Too many instance attributes (11/7) (too-many-instance-attributes) - **by design**
W: 62,15: Catching too general exception Exception (broad-except) - **by design**
E: 76,12: Raising NoneType while only classes or instances are allowed (raising-bad-type) - **dynamic code, just incorrect warning**
************* Module tabulator.helpers
W: 25, 0: TODO: rewrite without urlparse (fixme) - **todo**
W: 42, 0: FIXME: This supports valid paths like file://foo.csv, but also (fixme) - **todo**
W: 56, 0: TODO: does it work? (fixme) - **todo**
E:  9, 0: Unable to import 'six.moves.urllib.parse' (import-error) - **dynamic code, just incorrect warning**
W: 37, 8: Redefining built-in 'format' (redefined-builtin) - **reasonable but could be discussed**
W: 49,20: Redefining built-in 'bytes' (redefined-builtin) - **reasonable but could be discussed**
W: 78,11: Catching too general exception Exception (broad-except) - **by design**
C: 83, 8: Invalid variable name "e" (invalid-name) - **who said?**
W: 95,11: Catching too general exception Exception (broad-except) - **by design**
************* Module tabulator.table
C: 88, 0: Wrong hanging indentation (remove 4 spaces). - **as far I now recommended way 8 spaces here**
...
R: 18, 0: Too many arguments (12/5) (too-many-arguments) - **by design**
R: 18, 0: Too many local variables (17/15) (too-many-locals) - **by design**
R: 18, 0: Too many branches (13/12) (too-many-branches) - **by design**
...
R:  1, 0: Similar lines in 3 files - **by design**
...
and so on
```

With `pylint -E` (only errors) it fails with 3 false positives errors.

So it looks like we have to disable half of codes or migrate to other tool..

Here is discussion - http://jbisbee.blogspot.ru/2014/04/getting-started-with-pylint.html?showComment=1397032819394#c1157023345955046944 - the same thing `signal-to-noise` ratio is mentioned.

**Comment by pwalsh** [comment]
Hey. So, I get it looks annoying, but that is just a couple of things to add to our base `pylintrc`, and then in one or two places to add ignores in certain files. I'll update the `pylintrc`

**Comment by roll** [comment]
Thanks! I'm sure we find a way how to use it. But I suppose we also have to test we're not fighting against our tools. I've checked the problem a few days ago - there are two big  tools (around 400k uses on github of both libs) `pylint` and `flake8` (`pep8+pyflakes+mccabe`) and there is difference between them in concept. 

`pylint` in my opinion it's basic check (i don't know how it works exactly but for example it doesn't check for https://www.python.org/dev/peps/pep-0008/#inline-comments by default while `pep8` does) + set of subjective rules more like code quality check like https://codeclimate.com/ or https://www.quantifiedcode.com

`flake8` as combination of `pep8` and `pyflakes` is just explicit check against standard (pep8) and static analysis errors like bad import and unused variables.

So in my opinion `flake8` concept is more suitable to be part of CI/CD build to check out code follows standards and has no errors. And `pylint` concept is more suitable as code quality tool may be as part of reviewing when we don't need binary `green/red` but want to get grade of subjective code quality.

Just wanted to drop it somewhere because I've made some small discovery job (my interpretations could be wrong) - but basically I don't think it's an important question. We could use anything that works (like configured pylint) and then check how it's going on our real projects.

**Comment by roll** [comment]
@pwalsh 
I think we need to return to it - the whole python stack don't use any linting for now. It's kinda a dangerous situation. I think we need to enable it as part of `make test`.

I have a really good experience with `pylama` in OpenTrials repos - https://github.com/opentrials/collectors/blob/master/pylama.ini - easy to configure, all errors have codes etc The main difference with `pylint` - it's just syntax check + pep8 + complexity (if needed) - no subjective rules on how code should look.

Other option if considering `pylama` no production ready - `flake8` (pyflakes + pep8).

Option I prefer less - finally to config .pylintrc across the repos :)

**Comment by pwalsh** [comment]
@roll ok. go for `pylama` then.

**Comment by roll** [comment]
@pwalsh 
Thanks. I'll start to migrate for this approach gradually.
